SixTrack/sixtrack.s

```diff
@@ -41349,6 +41349,41 @@ C Should get me a NaN
          iexpr_dynk(niexpr_dynk+3) = 0 ! seed2 (current)
 
          niexpr_dynk = niexpr_dynk+3
+
+      case("RANDON")
+         ! RANDON: Turn by turn On for one turn with the probability P, else OFF
+         call dynk_checkargs(getfields_nfields,6,
+     &        "FUN funname RANDON seed1 seed2 P" )
+         call dynk_checkspace(4,1,1)
+	          
+         ! Set pointers to start of funs data blocks
+         nfuncs_dynk = nfuncs_dynk+1
+         niexpr_dynk = niexpr_dynk+1
+         nfexpr_dynk = nfexpr_dynk+1
+         ncexpr_dynk = ncexpr_dynk+1
+
+         ! Store pointers
+         funcs_dynk(nfuncs_dynk,1) = ncexpr_dynk !NAME (in cexpr_dynk)
+         funcs_dynk(nfuncs_dynk,2) = 8           !TYPE (RANDON)
+         funcs_dynk(nfuncs_dynk,3) = niexpr_dynk !seed1(initial), seed2(initial), seed1(current), seed2(current)
+         funcs_dynk(nfuncs_dynk,4) = nfexpr_dynk !ARG1 (P)
```

Comment:
This would be "P (in fexpr_dynk)", not "ARG1 (P)". The one below would be "ARG3 (unused)"